### PR TITLE
feat (react-email): Ignore test files when generating email previews

### PR DESCRIPTION
**Pull request by jwarykowski**
## Overview

This PR resolves an issue when generating emails previews, currently if you have any spec or test file within this directory these are picked up as they end in the `jsx` or `tsx` extension.

### CODE REVIEW

**Comment by gabrielmfern** [comment]
Just for curiosity, no judging, but in what situation did you need to write tests for your emails?

**Comment by jwarykowski** [comment]
> Just for curiosity, no judging, but in what situation did you need to write tests for your emails?

Hey @gabrielmfern, no worries totally fine. 
By implementing units tests it enables us to perform content validation, furthermore as the code evolves it gives us a safety net to catch any regressions that could be introduced. As the email is initially a react component I can test the output and any conditional logic just like I would for a normal application component.

Note this isn't 💯 but allows us to catch things earlier in the development cycle, I would also want to test the HTML generated via an email testing service (such as Litmus) or perform visual regression with a tool like Percy.

Personally where I can (and where it makes sense) I like to have 💯 unit test coverage, if you have critical applications or services I prefer the added confidence 😬😅

**Comment by jwarykowski** [comment]
@vcapretz / @gabrielmfern - Is there any chance this could be merged?

**Comment by gabrielmfern** [comment]
> @vcapretz / @gabrielmfern - Is there any chance this could be merged?

Hey, I am not a official maintainer of react-email so I can't really merge it. But just a heads up, as far as I know, the guys at Resend (the company behind react-email) are working really hard on their product, so to maintain react-email they can't iterate nor merge things really fast, at least not yet. Maybe it will take a bit until they can merge the PR, it doesn't mean they gave up on react-email nor stopped maintaining it, it's just at a slower pace rn.

**Comment by thesaltygerman** [comment]
I tired to wait. There is no pace just quiet. @jwarykowski @gabrielmfern I use jsx fork now with active. You should look at it.

**Comment by jwarykowski** [comment]
Hey @gabrielmfern, apologies and thanks for the feedback. No worries, I appreciate they can't always be available so all good. I have a simple workaround for this issue at the moment anyhow. 

@thesaltygerman No problems, I've already has the maintainer of jsx reach out to me and will try and find some time to evaluate it.

**Review by gabrielmfern — APPROVED** [review]

**Pull request merged**
